main.go

```diff
@@ -78,11 +83,28 @@ func main() {
 		log.Fatalf("watch error: %v", err)
 	}
 
-	fmt.Println(snc.Hash())
+	var cmd *exec.Cmd
+	cmd = checkPublish(cmd, snc.Hash(), true)
 
 	for hash := range snc.Events() {
+		cmd = checkPublish(cmd, hash, false)
+	}
+}
+
+func checkPublish(cmd *exec.Cmd, hash string, printLogs bool) *exec.Cmd {
+	var err error
+	if *publish && hash != "" {
+		cmd, err = shell.Publish(cmd, hash, *key)
+		if err != nil {
+			fmt.Printf("Publish error: %s\n", err.Error())
+			os.Exit(1)
+		} else if printLogs {
+			fmt.Printf("Publishing to key %s\n", *key)
+		}
+	} else if !*publish {
```

Comment:
this function should not be invoked if that's true

Reply:
checkPublish is function to keep main() clean and avoid repetition. This function checks if publish is true then publishes to IPNS else it just prints the hashes(Original functionality)

Reply:
as I mentioned, please use `if !*publish { go publisher }` and then in loop use `if !*publish { chan <- hash }`